lib/utils/RichEmbed.ts

```diff
@@ -170,7 +175,7 @@ export class RichEmbed {
             this.description === embed.description &&
             this.url === embed.url &&
             this.timestamp === embed.timestamp &&
-            this.fields.length === embed.fields.length &&
+            this.fields.length === (embed.fields ? embed.fields?.length : 0) &&
```

Comment:
I'll try to perform a proper fix on the `RichEmbed` class as this fix looks a bit off.